src/server/tiering/op_manager.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/inlined_vector.h>
+
+#include <variant>
+
+#include "server/tiering/common.h"
+#include "server/tiering/disk_storage.h"
+
+namespace dfly::tiering {
+
+// Manages READ/DELETE/STASH operations on top of a DiskStorage.
+// Implicitly combines reads with different offsets on the same 4kb page,
+// safely schedules deletes after reads and allows cancelling pending stashes
+class OpManager {
+ public:
+  // Two separate keyspaces are provided - one for strings, one for numeric identifiers.
+  // Ids can be used to track auxiliary values that don't map to real keys (like packed pages).
+  using EntryId = std::variant<unsigned, std::string_view>;
+  using OwnedEntryId = std::variant<unsigned, std::string>;
```

Comment:
OwnedEntryId can be private.

Reply:
Yes, maybe, I don't like how it's casted around currently either way. If it's private, I can't access it from outside functions